### PR TITLE
Fixed 'Orbot appears not to be running' (#1792)

### PR DESCRIPTION
**Pull request by vanitasvitae**
Hi!
I tried my best to fix the faulty "Orbot appears not to be running" message on Android 6 by ditching the old OrbotHelper.isOrbotRunning() method. I implemented a BroadcastReceiver that listens for Orbot Status intents and acts as a singleton with getters for the received states.

Im unsure, whether this is secure and proper implemented or not, so please take a close look at my changes :)

### CODE REVIEW

**Comment by dschuermann** [comment]
Looks good.
Does this work properly in your tests?

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/OrbotStatusReceiver.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+
+import org.sufficientlysecure.keychain.util.Log;
+
+/**
+ * Created by vanitas on 11.05.16.
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/OrbotStatusReceiver.java]: remove the "created by" line, instead add a license header like in other files with your name

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/OrbotStatusReceiver.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * BroadcastReceiver that receives Orbots status
+ */
+public class OrbotStatusReceiver extends BroadcastReceiver {
+
+    private static final String TAG = "OrbStatRec";
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/OrbotStatusReceiver.java]: use Constants.TAG instead

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/OrbotStatusReceiver.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    private int proxy_port_socks;
+
+    private static OrbotStatusReceiver instance;
+
+    public OrbotStatusReceiver() {
+        instance = this;
+    }
+
+    public static OrbotStatusReceiver getInstance() {
+        if(instance == null) instance = new OrbotStatusReceiver();
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/OrbotStatusReceiver.java]: if else always with {}

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/TorServiceUtils.java [hunk 4]
```diff
@@ -1,156 +0,0 @@
-/* This is the license for Orlib, a free software project to
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/orbot/TorServiceUtils.java]: great to get rid of this file :) I don't like it

**Comment by vanitasvitae** [comment]
I changed the points you mentioned. Im not sure what you meant with the authorship so I removed it completely.

I tested this manually on my device by deactivating/starting Orbot. Are there any test cases which I can use or should I write my own?

**Comment by dschuermann** [comment]
Great work!

**Comment by dschuermann** [comment]
Regarding tests: I don't think this can be tested in an automated manner.

**Comment by vanitasvitae** [comment]
Is seems like OpenKeyChain is checking Orbots status nearly every second. I'm going to take a look at this since it may drain battery.

**Pull request merged**